raft/node_test.go

```diff
@@ -728,3 +728,54 @@ func TestIsHardStateEqual(t *testing.T) {
 		}
 	}
 }
+
+func TestNodeProposeAddLearnerNode(t *testing.T) {
+	ticker := time.NewTicker(time.Millisecond * 100)
+	defer ticker.Stop()
+	n := newNode()
+	s := NewMemoryStorage()
+	r := newTestRaft(1, []uint64{1}, 10, 1, s)
+	go n.run(r)
+	n.Campaign(context.TODO())
+	stop := make(chan struct{})
+	done := make(chan struct{})
+	applyConfChan := make(chan struct{})
+	go func() {
+		defer close(done)
+		for {
+			select {
+			case <-stop:
+				return
+			case <-ticker.C:
+				n.Tick()
+			case rd := <-n.Ready():
+				s.Append(rd.Entries)
+				t.Logf("raft: %v", rd.Entries)
+				for _, ent := range rd.Entries {
+					if ent.Type == raftpb.EntryConfChange {
```

Comment:
indentation is crazy here. let us kill one indentation by doing 

```
if ent.Type != raftpb.EntryConfChange {
    continue
}
var cc raftpb.ConfChange
...
```

Reply:
+1